modules/trino/src/test/java/org/testcontainers/containers/TrinoContainerTest.java

```diff
@@ -0,0 +1,122 @@
+            }
+        }
+    }
+
+    @Test
+    public void testSpecificVersion() throws Exception {
+        try (TrinoContainer trino = new TrinoContainer(TrinoTestImages.TRINO_PREVIOUS_VERSION_TEST_IMAGE)) {
+            trino.start();
+            try (Connection connection = trino.createConnection();
+                 Statement statement = connection.createStatement();
+                 ResultSet resultSet = statement.executeQuery("SELECT DISTINCT node_version FROM system.runtime.nodes")) {
+                assertTrue("No result", resultSet.next());
+                assertEquals("Trino version", TrinoTestImages.TRINO_PREVIOUS_VERSION_TEST_IMAGE.getVersionPart(), resultSet.getString("node_version"));
+            }
+        }
+    }
+
+    @Test
+    public void testQueryMemoryAndTpch() throws SQLException {
+        try (TrinoContainer trino = new TrinoContainer(TrinoTestImages.TRINO_TEST_IMAGE)) {
+            trino.start();
+            try (Connection connection = trino.createConnection();
+                 Statement statement = connection.createStatement()) {
+                // Prepare data
+                statement.execute("CREATE TABLE memory.default.table_with_array AS SELECT 1 id, ARRAY[1, 42, 2, 42, 4, 42] my_array");
+
+                // Query Trino using newly created table and a builtin connector
+                try (ResultSet resultSet = statement.executeQuery("" +
+                    "SELECT nationkey, element " +
+                    "FROM tpch.tiny.nation " +
+                    "JOIN memory.default.table_with_array twa ON nationkey = twa.id " +
+                    "LEFT JOIN UNNEST(my_array) a(element) ON true " +
+                    "ORDER BY element OFFSET 1 FETCH NEXT 3 ROWS WITH TIES ")) {
+                    List<Integer> actualElements = new ArrayList<>();
+                    while (resultSet.next()) {
+                        actualElements.add(resultSet.getInt("element"));
+                    }
+                    Assert.assertEquals(Arrays.asList(2, 4, 42, 42, 42), actualElements);
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testInitScript() throws Exception {
+        try (TrinoContainer trino = new TrinoContainer(TrinoTestImages.TRINO_TEST_IMAGE)) {
+            trino.withInitScript("initial.sql");
+            trino.start();
+            try (Connection connection = trino.createConnection();
+                 Statement statement = connection.createStatement();
+                 ResultSet resultSet = statement.executeQuery("SELECT a FROM memory.default.test_table")) {
+                assertTrue("No result", resultSet.next());
+                assertEquals("Value", 12345678909324L, resultSet.getObject("a"));
+                assertFalse("Too many result", resultSet.next());
+            }
+        }
+    }
+
+    @Test
+    public void testTcJdbcUri() throws Exception {
```

Comment:
already tested in `TrinoJDBCDriverTest`, let's remove

Reply:
it tests more than TrinoJDBCDriverTest implicitly does, doesn't it?

still, can remove if you really want me to

Reply:
See https://github.com/testcontainers/testcontainers-java/pull/3668#discussion_r575241048